File: src/main/java/toast/pine/Classes/Skills/Skill.java
package toast.pine.Classes.Skills;

import toast.pine.Classes.CharacterClass;
import toast.pine.Classes.Skills.Interfaces.SkillInterface;

import java.util.List;

public abstract class Skill extends CharacterClass implements SkillInterface {

    protected final String skillName;
    protected final CharacterClass intendedClassUse;
    protected final SkillAOE skillAOE;
    protected final SkillDuration skillDuration;
    protected final List<String> skillDescription;
    protected final int skillLevel;
    protected final int skillMaxLevel;
    protected final int skillCost;
    protected final int skillCooldown;

    public Skill(String skillName, SkillAOE skillAOE, SkillDuration skillDuration, CharacterClass intendedClassUse, List<String> skillDescription, int skillLevel, int skillMaxLevel, int skillCost, int skillCooldown) {
        super(intendedClassUse.getName(), intendedClassUse.getIcon(), intendedClassUse.getHealth(), intendedClassUse.getDamage(), intendedClassUse.getMaxMana(), intendedClassUse.getManaPerSec(), intendedClassUse.getDefense(), intendedClassUse.getMaxLevel());
        this.skillName = skillName;
        this.skillAOE = skillAOE;
        this.skillDuration = skillDuration;
        this.intendedClassUse = intendedClassUse;
        this.skillDescription = skillDescription;
        this.skillLevel = skillLevel;
        this.skillMaxLevel = skillMaxLevel;
        this.skillCost = skillCost;
        this.skillCooldown = skillCooldown;
    }

    public String getSkillName() {
        return skillName;
    }

    public SkillAOE getSkillAOE() {
        return skillAOE;
    }

    public SkillDuration getSkillDuration() {
        return skillDuration;
    }

    public CharacterClass getIntendedClassUse() {
        return intendedClassUse;
    }

    public List<String> getSkillDescription() {
        return skillDescription;
    }

    public int getSkillLevel() {
        return skillLevel;
    }

    public int getSkillMaxLevel() {
        return skillMaxLevel;
    }

    public int getSkillCost() {
        return skillCost;
    }

    public int getSkillCooldown() {
        return skillCooldown;
    }
}

File: src/main/java/pine/toast/Classes/Skills/Skill.kt
package toast.pine.Classes.Skills

import toast.pine.Classes.CharacterClass

abstract class Skill(
    skillName: String,
    intendedClassUse: CharacterClass,
    skillDescription: List<String>,
    skillLevel: Int,
    skillMaxLevel: Int,
    skillCost: Int,
    skillCooldown: Int
) : CharacterClass(
    intendedClassUse.getName(),
    intendedClassUse.getIcon(),
    intendedClassUse.getHealth().toDouble(),
    intendedClassUse.getDamage(),
    intendedClassUse.getMaxMana(),
    intendedClassUse.getManaPerSec(),
    intendedClassUse.getDefense(),
    intendedClassUse.getMaxLevel()
) {
    private val skillName: String
    private val intendedClassUse: CharacterClass
    private val skillDescription: List<String>
    private var skillLevel: Int
    private val skillMaxLevel: Int
    private var skillCost: Int
    private var skillCooldown: Int

    init {
        this.skillName = skillName
        this.intendedClassUse = intendedClassUse
        this.skillDescription = skillDescription
        this.skillLevel = skillLevel
        this.skillMaxLevel = skillMaxLevel
        this.skillCost = skillCost
        this.skillCooldown = skillCooldown
    }

    fun getIntendedClassUse(): CharacterClass {
        return intendedClassUse
    }

    fun getSkillDescription(): List<String> {
        return skillDescription
    }

    fun getSkillLevel(): Int {
        return skillLevel
    }

    fun getSkillMaxLevel(): Int {
        return skillMaxLevel
    }

    fun setSkillLevel(level: Int) {
        skillLevel = level
    }

    fun getSkillCost(): Int {
        return skillCost
    }

    fun setSkillCost(cost: Int) {
        skillCost = cost
    }

    fun getSkillCooldown(): Int {
        return skillCooldown
    }

    fun setSkillCooldown(cooldown: Int) {
        skillCooldown = cooldown
    }

    abstract fun getSkill(): Skill
    abstract fun getSkillName(): String






}
